storage/inkless/src/main/resources/db/migration/V5__Fix_deadlock_in_delete_records_v1.sql

```diff
@@ -0,0 +1,90 @@
+-- Copyright (c) 2025 Aiven, Helsinki, Finland. https://aiven.io/
+
+CREATE OR REPLACE FUNCTION delete_records_v1(
+    arg_now TIMESTAMP WITH TIME ZONE,
+    arg_requests delete_records_request_v1[]
+)
+RETURNS SETOF delete_records_response_v1 LANGUAGE plpgsql VOLATILE AS $$
+DECLARE
+    l_request RECORD;
+    l_log RECORD;
+    l_converted_offset BIGINT = -1;
+    l_deleted_bytes BIGINT;
+BEGIN
+
+    DROP TABLE IF EXISTS affected_files;
+    CREATE TEMPORARY TABLE affected_files (
+        file_id BIGINT PRIMARY KEY
+    )
+    ON COMMIT DROP;
+
+    FOR l_request IN
+        SELECT *
+        FROM unnest(arg_requests)
+        ORDER BY topic_id, partition  -- ordering is important to prevent deadlocks
```

Comment:
This line is the main change in this PR, moved from the `SELECT` below. The rest of the function is unchanged.